Thumbnail/FormatThumbnail.php

```diff
@@ -84,18 +85,59 @@ public function generate(MediaProviderInterface $provider, MediaInterface $media
     /**
      * {@inheritdoc}
      */
-    public function delete(MediaProviderInterface $provider, MediaInterface $media)
+    public function delete(MediaProviderInterface $provider, MediaInterface $media, $format = null)
     {
-        // delete the differents formats
-        foreach ($provider->getFormats() as $format => $definition) {
-            $path = $provider->generatePrivateUrl($media, $format);
-            if ($path && $provider->getFilesystem()->has($path)) {
-                $provider->getFilesystem()->delete($path);
+        $filesystem = $provider->getFilesystem();
+
+        if (is_null($format)) {
+            //delete all thumbnails
+            $format = 'default_whateverformat';
+
+            $relPath = $provider->generatePrivateUrl($media, $format);
+            $relDir = dirname($relPath);
+            $fsRootDir = $this->getFilesystemRootDir($filesystem);
```

Comment:
I mean if there was some format long time ago, and i do not remember it's name, I cannot remove it, so code should search filesystem by mask.